OneSignalSDK/onesignal/core/src/main/java/com/onesignal/internal/OneSignalImp.kt

```diff
@@ -357,12 +358,16 @@ internal class OneSignalImp : IOneSignal, IServiceProvider {
             currentIdentityOneSignalId = identityModelStore!!.model.onesignalId
 
             if (currentIdentityExternalId == externalId) {
+                // login is for same user that is already logged in, fetch (refresh)
+                // the current user.
+                identityModelStore!!.model.jwtToken = jwtBearerToken
```

Comment:
Yes, we don't want to broadcast the event twice because the identity model will be replaced by a new one if a different user is logged in, so we don't update the jwt for the current user but only the new user.

Reply:
I see, thanks for the details, marking as resolved.